src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs

```diff
@@ -77,6 +77,11 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(
         using var request = new HttpRequestMessage(HttpMethod.Get, address);
         request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue(OpenIdAuthenticationConstants.Media.Json));
 
+        // Add referer and origin header so steam does not 403 the request
+        // Started to be a requirement on the 12th of February 2025
+        Options.Backchannel.DefaultRequestHeaders.Add("referer", "https://steamcommunity.com");
```

Comment:
You can't do that: the `Backchannel` is shared across requests so you can't mutate it safely.

Reply:
➕ - you'd need to add them to the `HttpRequestMessage` instead.